lib/src/storage/backend/rocksdb.rs

```diff
@@ -647,8 +753,25 @@ impl Drop for Reader {
     }
 }
 
-impl Reader {
-    pub fn get(
+pub trait ReaderTrait {
+    fn get(
+        &self,
+        column_family: &ColumnFamily,
+        key: &[u8],
+    ) -> Result<Option<PinnableSlice>, StorageError>;
+    fn contains_key(&self, column_family: &ColumnFamily, key: &[u8]) -> Result<bool, StorageError>;
+    fn iter(&self, column_family: &ColumnFamily) -> Result<Iter, StorageError>;
+    fn scan_prefix(
+        &self,
+        column_family: &ColumnFamily,
+        prefix: &[u8],
+    ) -> Result<Iter, StorageError>;
+    fn len(&self, column_family: &ColumnFamily) -> Result<usize, StorageError>;
+    fn is_empty(&self, column_family: &ColumnFamily) -> Result<bool, StorageError>;
+}
```

Comment:
Instead of a trait we might embedd an enum instead of the `Reader` struct just like what we do for `DbHandler`. It might allow to avoid some code duplication.